integration-tests/tests/common.rs

```diff
@@ -0,0 +1,308 @@
+            query_pairs.append_pair("secretId", &self.secret_id);
+
+            if let Some(version_id) = &self.version_id {
+                query_pairs.append_pair("versionId", version_id);
+            }
+
+            if let Some(version_stage) = &self.version_stage {
+                query_pairs.append_pair("versionStage", version_stage);
+            }
+
+            if let Some(refresh_now) = self.refresh_now {
+                query_pairs.append_pair("refreshNow", &refresh_now.to_string());
+            }
+        }
+        url.query().unwrap_or("").to_string()
+    }
+}
+
+pub struct AgentProcess {
+    pub child: tokio::process::Child,
+    pub port: u16,
+}
+
+impl Drop for AgentProcess {
+    fn drop(&mut self) {
+        let _ = self.child.start_kill();
+    }
+}
+
+pub async fn start_agent_on_port(port: u16) -> AgentProcess {
+    let config_content = format!(
+        r#"
+http_port = {}
+log_level = "info"
+ttl_seconds = 5
+cache_size = 100
+validate_credentials = false
+"#,
+        port
+    );
+
+    let config_path = format!("/tmp/test_config_{}.toml", port);
+    std::fs::write(&config_path, config_content).expect("Failed to write test config");
+
+    env::set_var("AWS_TOKEN", "test-token-123");
+
+    let possible_paths = [
+        PathBuf::from("target")
+            .join("release")
+            .join("aws_secretsmanager_agent"),
+        PathBuf::from("target")
+            .join("debug")
+            .join("aws_secretsmanager_agent"),
+        PathBuf::from("..")
+            .join("target")
+            .join("release")
+            .join("aws_secretsmanager_agent"),
+        PathBuf::from("..")
+            .join("target")
+            .join("debug")
+            .join("aws_secretsmanager_agent"),
+    ];
+
+    let agent_path = possible_paths
+        .iter()
+        .find(|path| path.exists())
+        .expect("Agent binary not found");
+
+    let mut child = TokioCommand::new(agent_path)
+        .arg("--config")
+        .arg(&config_path)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn()
+        .expect("Failed to start agent");
+
+    // Read stdout until we see the "listening" message
+    let stdout = child.stdout.take().expect("Failed to get stdout");
+    let mut reader = BufReader::new(stdout).lines();
+
+    match reader.next_line().await {
+        Ok(Some(line)) => {
+            if !line.contains("listening on") {
+                let _ = child.kill().await;
+                panic!("Agent failed to start - no listening message found");
+            }
+        }
+        Ok(None) => {
+            let _ = child.kill().await;
+            panic!("Stream ended without finding listening message");
+        }
+        Err(e) => {
+            let _ = child.kill().await;
+            panic!("Failed to read agent output: {}", e);
+        }
+    }
+
+    AgentProcess { child, port }
+}
+
+pub async fn make_agent_request(port: u16, query: &AgentQuery) -> String {
```

Comment:
I wrote this below in the other file, this should be a method of `AgentProcess`.